internal/pkg/cli/env_upgrade.go

```diff
@@ -0,0 +1,72 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package cli
+
+import (
+	"fmt"
+
+	"github.com/spf13/cobra"
+)
+
+// envUpgradeVars holds flag values.
+type envUpgradeVars struct {
+	*GlobalOpts
+
+	name string // Required. Name of the environment.
+	all  bool   // True means all environments should be upgraded.
```

Comment:
Does `all` supersede `name`? For example, will we require a customer to run `copilot env upgrade --all --name test` if they want to upgrade everything?

Reply:
Yeah that was my thinking too, so `copilot env upgrade --all` to upgrade everything. They won't be able to use `--name` and `--all` together, `--name` will be to upgrade only a specific env.